lib/enum/nested.rb

```diff
@@ -18,21 +18,27 @@ def initialize(people_array)
   end
 
   def ages_sum
+    @people.map { |person| person.age}.inject(:+)
   end
 
   def average_age
+    @people.map { |person| person.age}.inject(:+) / @people.count
   end
 
   def total_years_programming_experience_for_all_languages
+     @people.flat_map { |person| person.years_language_experience.values}.inject(:+)
   end
 
   def favorite_food_frequency
+    @people.flat_map { |person| person.favorite_foods}.each_with_object(Hash.new(0)) { |count, food| food[count] += 1 }
   end
 
   def total_combined_years_language_experience(language)
+    @people.flat_map { |person| person.years_language_experience[language]}.compact!.inject(:+)
```

Comment:
I don't think you need `compact!` here, since the non-mutating version `compact` will do what you want and will never return `nil` whereas `compact!` will sometimes return `nil` (if there are no duplicates).